### PR TITLE
Update CRAM conversion and round trip tests.

### PR DESCRIPTION
**Pull request by cmnbroad**
The existing CRAM compliance tests do only partial record validation after conversion, rather than testing for full record equality. This PR separates out the tests that fail if full record validation is used (tickets filed) and:
-changes the remaining tests to require full SAMRecord equality.
-Adds a BAM->CRAM->BAM round trip validation test based on an NA12878 subset.
-Removes a couple of unreferenced or duplicate alignment and reference files.
### Checklist
- [x] Code compiles correctly
- [ ] New tests covering changes and new functionality
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

File: src/test/java/htsjdk/samtools/CRAMComplianceTest.java [hunk 1]
```diff
@@ -147,4 +175,156 @@ private void assertSameRecords(int majorVersion, SAMRecord record1, SAMRecord re
...
+        for (int i = 0; i < originalBAMRecords.size(); i++) {
+            Assert.assertEquals(originalBAMRecords.get(i), cramRecords.get(i));
+        }
+
+        //  write the roundtripped CRAM records out to a bam and compare to the original
+        final File roundTrippedBAMFile = File.createTempFile("testBAMThroughCRAMRoundTrip", BamFileIoUtils.BAM_FILE_EXTENSION);
+        roundTrippedBAMFile.deleteOnExit();
+        writeRecordsToFile(cramRecords, roundTrippedBAMFile, referenceFile, samHeader);
+
+        Assert.assertTrue(compareMD5Digests(roundTrippedBAMFile, originalBAMInputFile));
```
Comment (vadimzalunin) [inline src/test/java/htsjdk/samtools/CRAMComplianceTest.java]: not sure about binary equality of BAM files, for example changing compression level will break this.

File: src/test/java/htsjdk/samtools/CRAMComplianceTest.java [hunk 2]
```diff
@@ -147,4 +175,156 @@ private void assertSameRecords(int majorVersion, SAMRecord record1, SAMRecord re
...
+            }
+        }
+        return recs;
+    }
+
+    private void writeRecordsToFile (
+            final List<SAMRecord> recs,
+            final File targetFile,
+            final File referenceFile,
+            final SAMFileHeader samHeader)  throws IOException {
```
Comment (vadimzalunin) [inline src/test/java/htsjdk/samtools/CRAMComplianceTest.java]: remove 'throws IOException'

File: src/test/java/htsjdk/samtools/CRAMComplianceTest.java [hunk 3]
```diff
@@ -147,4 +175,156 @@ private void assertSameRecords(int majorVersion, SAMRecord record1, SAMRecord re
...
+            throws NoSuchAlgorithmException, IOException {
+        MessageDigest rtMD = MessageDigest.getInstance("MD5");
+        MessageDigest origMD = MessageDigest.getInstance("MD5");
+
+        String origMD5;
+        String rtMD5;
+        try (FileInputStream rtStream = new FileInputStream(tempRoundTrippedBAMFile);
+            DigestInputStream rtDigestStream = new DigestInputStream(rtStream, rtMD);
+            FileInputStream origStream = new FileInputStream(tempOriginalBAMFile);
+            DigestInputStream origDigestStream = new DigestInputStream(origStream, origMD);) {
```
Comment (vadimzalunin) [inline src/test/java/htsjdk/samtools/CRAMComplianceTest.java]: remove semicolumn

**Comment by vadimzalunin** [comment]
summoning @droazen @yfarjoun @cmnbroad into this thread
It appears I'm blocked on this PR... Given the changes I've requested are not important, can we just merge this?

**Comment by cmnbroad** [comment]
Its fine with me - this only has test changes.

**Comment by cmnbroad** [comment]
I'm not sure why the codecov report shows a change in src/main/java/htsjdk/samtools/util/AsyncBufferedIterator.java that looks unrelated to this PR, but is looks to be timing related. The tests are passing, so I"m merging.

**Pull request merged**